### PR TITLE
feat(docker): added base dockerfile

### PR DESCRIPTION
**Pull request by shanduur**
## What this PR does / why we need it:

Initial Docker configuration. Cherry-picked from #1 .

## Which issue(s) this PR resolves:

N/A

## Special notes for your reviewer:

N/A

## Additional documentation e.g., enhancement proposals, usage docs, etc.:

N/A

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+# Build.
+RUN make build
+
+#########################################################################################
+# Runtime
+#########################################################################################
+
+# Second stage: building final environment for running the executable.
+FROM gcr.io/distroless/static-debian11:latest AS runtime
+
```
Comment (avestuk) [inline Dockerfile:52]: Is it possible to run the driver as a non root user? 

If so `USER 65532:65532` is good.

**Review by avestuk — DISMISSED** [review]

File: Dockerfile (hunk 1 above)
Comment (shanduur) [reply Dockerfile:52]: Well, yes and no. I don't have much experience with the distroless images, and don't really know how to create `/var/lib/cosi` so the driver has enough permissions to create a socket there.

Comment (shanduur) [reply Dockerfile:56]: The last commit should works as expected (container starts and binds a socket), with the final user being (as you mentioned) `USER 65532:65532`.

**Review by avestuk — APPROVED** [review]

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+# Set the final UID:GID to non-root user.
+USER 65532:65532
+
+# Disable healthcheck.
+HEALTHCHECK NONE
+
+# Add labels
+LABEL name="linode-cosi-driver"
+LABEL description="COSI Driver for Linode Object Storage"
+LABEL vendor="Linode, LLC"
```
Comment (luthermonson) [inline Dockerfile:75]: this should be `Akamai Technologies, Inc.`

File: helm/linode-cosi-driver/values.yaml [hunk 3]
```diff
@@ -0,0 +1,98 @@
+# Default values for linode-cosi-driver.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+replicaCount: 1
+
+image:
+  repository: nginx
```
Comment (luthermonson) [inline helm/linode-cosi-driver/values.yaml:8]: i know none of this works yet, but probably dont want this to be nginx?

Comment (shanduur) [reply helm/linode-cosi-driver/values.yaml:8]: This is template generated by `helm create`, with manifests removed. I thought leaving the values will be a good starting point.

**Review by luthermonson — APPROVED** [review]

**Pull request merged**